src/backend/model_deployments/base.py

```diff
@@ -14,16 +16,51 @@ class BaseDeployment:
     list_models: List[str]: List all models.
     is_available: bool: Check if the deployment is available.
     """
+    @classmethod
+    def id(cls) -> str:
+        return cls.name().replace(" ", "_").lower()
 
-    @property
+    @classmethod
     @abstractmethod
-    def rerank_enabled(self) -> bool: ...
+    def name(cls) -> str: ...
 
-    @staticmethod
-    def list_models() -> List[str]: ...
+    @classmethod
+    @abstractmethod
+    def env_vars(cls) -> List[str]: ...
+
+    @classmethod
+    @abstractmethod
+    def rerank_enabled(cls) -> bool: ...
+
+    @classmethod
+    @abstractmethod
+    def list_models(cls) -> List[str]: ...
+
+    @classmethod
+    @abstractmethod
+    def is_available(cls) -> bool: ...
+
+    @classmethod
+    def is_community(cls) -> bool:
+        return False
+
+    @classmethod
+    def config(cls) -> Dict[str, Any]:
+        config = Settings().safe_lookup("deployments", cls.id(), default={})
+        return config.dict() if config else {}
 
-    @staticmethod
-    def is_available() -> bool: ...
+    @classmethod
+    def to_deployment_info(cls) -> DeploymentInfo:
+        data = {
```

Comment:
Can just

```
return DeploymentInfo(
    field=value,
    field2=value2,
..
)
```

Reply:
Also - if its not too much friction can you rename `DeploymentInfo` to `DeploymentDefinition`? To maintain consistency with the Tool config schema